core/src/main/java/org/wso2/carbon/kernel/utils/Utils.java

```diff
@@ -156,4 +173,207 @@ public static void checkSecurity() {
+        try {
+            properties.load(inputStream);
+            stringBuilder = new StringBuilder();
+            properties.entrySet().forEach(entry -> stringBuilder
+                    .append(createXmlElement(entry.getKey().toString(), entry.getValue().toString())));
+
+        } catch (IOException e) {
+            logger.error("Exception occurred while converting Properties to XML: ", e);
+            throw new RuntimeException("Exception occurred while converting Properties to XML: ", e);
+        }
+        String xmlString = createXmlElement(rootElement, stringBuilder.toString());
+        //Need to add a root element
+        return prettyFormatXMLString(xmlString);
+    }
+
+    /**
+     * This method converts a given XML String to Properties format
+     *
+     * @param xmlString XML String that needs to be converted to Properties format
+     * @return String in Properties format
+     */
+    public static String convertXMLToProperties(String xmlString, String rootElement) {
+        String jsonString = convertXMLToJSON(xmlString);
+        Yaml yaml = new Yaml();
+        Map map = yaml.loadAs(jsonString, Map.class);
+        //Remove root element
+        if (map.containsKey(rootElement)) {
+            map = (Map) map.get(rootElement);
+        } else {
+            throw new RuntimeException("Root element not found when converting XML to Properties");
+        }
+        StringBuilder stringBuilder = new StringBuilder();
+        String tempString;
+        for (Object entryObject : map.entrySet()) {
+            Map.Entry entry = (Map.Entry) entryObject;
+            tempString = entry.getKey() + " = " + entry.getValue();
+            stringBuilder.append(tempString).append("\n");
+        }
+        return stringBuilder.toString();
+    }
+
+    /**
+     * This method properly formats a given XML String
+     *
+     * @param xmlString XML String that needs to be formatted
+     * @return Properly formatted XML formatted String
+     */
+    public static String prettyFormatXMLString(String xmlString) {
+        Source xmlInput = new StreamSource(new StringReader(xmlString));
+        return convertXMLSourceToString(xmlInput);
+    }
+
+    /**
+     * This method converts the given Document to String format
+     *
+     * @param doc Document that needs to be converted to XML formatted String
+     * @return XML formatted String
+     */
+    public static String convertXMLtoString(Document doc) {
+        DOMSource domSource = new DOMSource(doc);
+        return convertXMLSourceToString(domSource);
+    }
+
+    /**
+     * This method converts the given Source to XML formatted String
+     *
+     * @param source Source that needs to be converted to XML formatted String
+     * @return XML formatted String
+     */
+    public static String convertXMLSourceToString(Source source) {
+        String xmlString;
+        try {
+            StringWriter stringWriter = new StringWriter();
```

Comment:
This is fixed in the following PR.

https://github.com/wso2/carbon-kernel/pull/1161